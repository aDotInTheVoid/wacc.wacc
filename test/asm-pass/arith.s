.intel_syntax noprefix
.global main
main:
    push rbp
    mov rbp, rsp
    sub rsp, 80
    mov eax, 1
    push rax # e_push_number
    mov eax, 1
    push rax # e_push_number
    pop rax
    pop rbx
    add ebx, eax
    push rbx
    pop rdi # load print
    call waccrt_println_i32
    mov eax, 100
    push rax # e_push_number
    mov eax, 1
    push rax # e_push_number
    pop rax
    pop rbx
    sub ebx, eax
    push rbx
    pop rdi # load print
    call waccrt_println_i32
    mov eax, 1
    push rax # e_push_number
    mov eax, 100
    push rax # e_push_number
    pop rax
    pop rbx
    sub ebx, eax
    push rbx
    pop rdi # load print
    call waccrt_println_i32
    mov eax, 2147483647
    push rax # e_push_number
    mov eax, 3
    push rax # e_push_number
    pop rax
    pop rbx
    add ebx, eax
    push rbx
    pop rdi # load print
    call waccrt_println_i32
    xor rax, rax
.ret_main:
    add rsp, 80
    pop rbp
    ret
.section .rodata

