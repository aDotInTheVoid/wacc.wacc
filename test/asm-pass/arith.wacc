begin 

println 1 + 1;
println 100 - 1;
println 1 - 100;
println 2147483647 + 3

end