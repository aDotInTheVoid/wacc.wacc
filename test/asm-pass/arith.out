2
99
-99
-2147483646
